.github/workflows/build.yml

```diff
@@ -15,90 +15,160 @@
 #
 name: Elyra Tests
 on:
-  push:
-    branches: '*'
-  pull_request:
-    branches: '*'
+  push: # all branches
+  pull_request: # all branches
   schedule:
     # once a day at 3 am (UTC) (7 pm (PST))
     - cron:  '0 3 * * *'
 
+env:
+  FORCE_COLOR: true
+
 jobs:
-  build:
-    runs-on: ${{ matrix.os }}
+  prepare-yarn-cache:
+    name: Prepare Cache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "*"
+      - uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+            /home/runner/.cache/Cypress
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+      - name: Install
+        run: yarn install --frozen-lockfile
+
+  lint-server:
+    name: Lint Server
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+      - name: Lint
+        run: flake8 elyra
+
+  lint-ui:
+    name: Lint UI
+    needs: prepare-yarn-cache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "*"
+      - uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+      - name: Install
+        run: yarn install
+      - name: Lint
+        run: yarn eslint:check --max-warnings=0
```

Comment:
same as above.

Reply:
We only have a make target `lint-ui` which runs both `eslint` and `prettier` together. I think it would be beneficial to run these in separate steps so it's easier to see what failed. Additionally, `lint-ui` runs both commands in `fix` mode instead of `check` mode. In CI we should also set `max-warnings` to zero so it will fail on any lint warnings/errors. 

Should I make two new make targets intended for checking?

Reply:
But wouldn't you see in the log which task failed (e.g. eslint vs pretier)?

Reply:
Yea, you could look in the log to see which task failed, but it’s a bit clearer if you separate the commands into separate steps because then the log output for both commands are separated in the UI.

You can tell much quicker what went wrong

![B7319152-5088-45FB-A7AE-4D5A9F196F9A](https://user-images.githubusercontent.com/4212769/119499773-0af9ee00-bd35-11eb-8b9d-741e6795ce3e.jpeg)